hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetUtils.java

```diff
@@ -229,6 +230,23 @@ public static InetSocketAddress createSocketAddr(String target,
     }
     target = target.trim();
     boolean hasScheme = target.contains("://");
+    if (StringUtils.countMatches(target, ":") > 2) {
+      // if scheme exists in the target, for example:
+      // https://ffff:ffff:ffff:ffff::1:XXXX will be formed like
+      // https://[ffff:ffff:ffff:ffff::1]:XXXX
```

Comment:
The code looks fine, we'd better use a specific port in the comments, such as `8088`

Reply:
Thanks for review.
modified as you mentioned.
Pls help to merge once the pipeline is down.
And pls help to review the other PR I raised if you got time.
Thanks a lot.